### PR TITLE
chore: Update MDC Web to v0.44.0

### PR DESCRIPTION
**Pull request by abhiomkar**
MDC Web v0.44.0 [changelog](https://github.com/material-components/material-components-web/blob/master/CHANGELOG.md#0440-2019-02-04).

### CODE REVIEW

File: src/TextFieldCatalog.js [hunk 1]
```diff
@@ -7,56 +7,84 @@ import ReactGA from 'react-ga';
...
+        : null
+      }
     </div>
   );
 }
 
-const FullWidthTextField = ({textarea, textFieldId, helperText}) => {
-  const classes = classnames('mdc-text-field', 'text-field', 'mdc-text-field--fullwidth', {
+const FullWidthTextField = ({textarea, textFieldId, helperText, characterCounter}) => {
+  // TODO: Remove --no-label class name in next release.
```
Comment (kfranqueiro) [inline src/TextFieldCatalog.js]: Just checking, but I assume this has to do with the bug we fixed after 0.44.0?

File: src/TextFieldCatalog.js [hunk 2]
```diff
@@ -7,56 +7,84 @@ import ReactGA from 'react-ga';
...
 }
 
-const Outline = ({textFieldId}) => (
+const Outline = ({textFieldId, noLabel}) => (
   <div className='mdc-notched-outline' key='outline'>
     <div className='mdc-notched-outline__leading'></div>
-    <div className='mdc-notched-outline__notch'>
-      <Label textFieldId={textFieldId}/>
-    </div>
+    {noLabel ?
```
Comment (kfranqueiro) [inline src/TextFieldCatalog.js]: Shouldn't we still be rendering the notch element, just not the label inside?

File: src/TextFieldCatalog.js [hunk 3]
```diff
@@ -67,20 +95,24 @@ const Label = ({textFieldId}) => (
   </label>
 );
 
-const Input = ({placeholder, textFieldId, onClick}) => (
+const Input = ({placeholder, textFieldId, onClick, noLabel, ...props}) => (
```
Comment (kfranqueiro) [inline src/TextFieldCatalog.js]: What was props added for in these? It's hard for me to tell offhand.

File: config/webpack.styles.config.js [hunk 4]
```diff
@@ -74,6 +74,9 @@ const scssLoaders = Object.assign(
         {
           loader: 'sass-loader',
           options: {
+            sourceMap: true,
```
Comment (kfranqueiro) [inline config/webpack.styles.config.js]: I'm seeing a bunch of logs to the effect of "Previous source map found, but options.sourceMap isn't set" from PostCSS Loader. Not sure if this is related?

File: src/TextFieldCatalog.js (hunk 1 above)
Comment (abhiomkar) [reply src/TextFieldCatalog.js]: That's correct. It was fixed in https://github.com/material-components/material-components-web/pull/4385 (after v0.44.0 release)

File: src/TextFieldCatalog.js (hunk 2 above)
Comment (abhiomkar) [reply src/TextFieldCatalog.js]: Notch element isn't necessary. See outlined example [without label](https://github.com/material-components/material-components-web/tree/master/packages/mdc-textfield#outlined-1) here.

File: src/TextFieldCatalog.js (hunk 3 above)
Comment (abhiomkar) [reply src/TextFieldCatalog.js]: Removed the use of `...props` and added `maxLength` attribute explicitly.

File: config/webpack.styles.config.js (hunk 4 above)
Comment (abhiomkar) [reply config/webpack.styles.config.js]: Not sure, are you seeing this in compile time?

Comment (kfranqueiro) [reply config/webpack.styles.config.js]: It seems to be at runtime when I load the catalog.

**Pull request merged**